themes/bootstrap3/js/common.js

```diff
@@ -38,13 +38,30 @@ var VuFind = (function VuFind() {
       }
     }
   };
+
+  var initDisableSubmitOnClick = function initDisableSubmitOnClick() {
+    $('[data-disable-on-submit]').on('submit', function handleOnClickDisable() {
+      var $form = $(this);
+      // Disable submit elements via setTimeout so that the submit button value gets
+      // included in the submitted data before being disabled:
+      setTimeout(
+        function disableSubmit() {
+          $form.find('[type=submit]').prop('disabled', true);
+        },
+        0
+      );
```

Comment:
Opinionated request: Instead of `setTimeout(fn, 0)`, could we use `requestAnimationFrame`? It accomplishes the same thing but feels less hacky to me.

Reply:
@crhallberg I don't have a preference, but from reading the spec for requestAnimationFrame it sounds like it's more real-time and would probably fire before next paint (which would be nice), but is it certain that the form submission has completed at that time? In other words, is it possible that requestAnimationFrame gets completed before initDisableSubmitOnClick returns?

Reply:
Not to my understanding. It uses the event loop to defer same as setTimeout.

Reply:
I'd rather be cautious here as well. Using setTimeout is the way recommended all over the web. I didn't find examples using requestAnimationFrame for this purpose. If you have convincing resources, please share. :)

Reply:
I found [this article](https://javascript.plainenglish.io/better-understanding-of-timers-in-javascript-settimeout-vs-requestanimationframe-bf7f99b9ff9b) fairly informative, though it doesn't really persuade me one way or the other in the setTimeout vs. requestAnimationFrame debate for this particular use case. It does bring up the interesting point that it's a good idea to clear timers that you have created (whichever method you use); I suspect VuFind has a lot of shortcomings on that point. Any thoughts on that, or anything else?

In any case, perhaps the path of least resistance would be to merge this as-is, and then create a separate PR to evaluate our use of timers and optimize/improve where feasible. I confess that I really hate setTimeout, as it almost always feels like a hack, but I recognize that it is often needed for legitimate reasons. :-) For now, as a starting point, I'll run the full test suite on this code as-is and report if it encounters any problems.

Reply:
The article still doesn't tell if it's possible that the callback get called before the form is submitted. I'd suppose it is, if it's called whenever the browser is ready to paint, and e.g. button release will require a paint.

Reply:
The talk in the article is one of the sources for learning about this but I didn't come away with any clearer on `requestAnimationFrame` so I do agree that `setTimeout` is the safer option. My tests show that requestAnimationFrame would work the same.

An even safer and clearer option might be to attach the listener to the btns and add a hidden input to the form to cover the disabling. I messed around with some JS and came up with this proposal:

```JS
$('[data-disable-on-submit]').each(function (i, form) {
  let $form = $(form);
  let $btns = $form.find("[type='submit']");

  $btns.on("click", function submit(e) {
    let $btn = $(e.target);

    let $input = $('<input type="hidden"/>');
    $input.attr("name", $btn.name);
    $input.val($btn.value);
    $form.append($input);

    $btns.prop("disabled", true);
    $form.submit(); // I don't like this part
  });
});
```

The `setTimeout` above is definitely more eloquent.

Reply:
Thanks, @crhallberg! Sounds like we can move forward with the existing solution, though future exploration may be worthwhile. I'll wait to hear what @EreMaijala thinks about merging now vs. doing more development before taking further action.

Reply:
@demiankatz I'd move forward with the existing solution. We can always revisit this. 

@crhallberg There's a potential problem with attaching to the buttons in that if something, such as late validation, blocks the form submit, the buttons get disabled prematurely and would prevent submitting the form after correction.

Reply:
Great, merging now! I welcome a follow-up PR to improve the solution and/or extend it to more forms, if and when there's a need.